### PR TITLE
update to sdk version 0.6.5 and change construction.

### PR DESCRIPTION
**Pull request by breedx-nr**
This brings in the new SDK version and gets rid of the deprecated style of construction.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -32,7 +32,8 @@ googleJavaFormat {
 
 dependencies {
     api("io.micrometer:micrometer-core:1.3.3")
-    api("com.newrelic.telemetry:telemetry:0.4.0")
+    api("com.newrelic.telemetry:telemetry:0.6.1")
+    api("org.slf4j:slf4j-api:1.7.30")
```
Comment (jkwatson) [inline build.gradle.kts:36]: What's the motivation for this change? I'm only concerned with pinning the version here as a `api` declaration and what impact that could have on consumers of the library.

Comment (breedx-nr) [reply build.gradle.kts:36]: I think that previously it must have been picked up as transitive from the SDK before or something, because as soon as I upgraded to `0.6.1` it broke compilation.  So I added a dependency here on slf4j directly (since it is used in the registry).  

What else can we do?

File: src/main/java/io/micrometer/newrelic/NewRelicRegistry.java [hunk 2]
```diff
@@ -263,22 +263,30 @@ public NewRelicRegistry build() {
...
-          throw new RuntimeException("Invalid URI for the metric API : " + config.uri(), e);
+      try {
+        URI uri = URI.create(config.uri());
+        if (uri.getPath() != null && uri.getPath().length() > 0) {
+          return builder.endpointWithPath(uri.toURL());
         }
+        return builder.endpoint(uri.getScheme(), uri.getHost(), uri.getPort());
+      } catch (MalformedURLException e) {
+        throw new RuntimeException("Invalid URI for the metric API : " + config.uri(), e);
       }
```
Comment (jkwatson) [inline src/main/java/io/micrometer/newrelic/NewRelicRegistry.java:289]: Given I've seen these lines of code twice in 2 days, in two different projects, should we consider making this hunk of logic part of the SDK configuration API directly?

Comment (breedx-nr) [reply src/main/java/io/micrometer/newrelic/NewRelicRegistry.java:289]: Absolutely!  My thoughts exactly.

Comment (breedx-nr) [reply src/main/java/io/micrometer/newrelic/NewRelicRegistry.java:289]: I'll put an issue in for that.

File: build.gradle.kts (hunk 1 above)
Comment (jkwatson) [reply build.gradle.kts:36]: Can we make it `implementation`, rather than `api`?

Comment (breedx-nr) [reply build.gradle.kts:36]: Sure thing.
